src/System.IO.Ports/src/System/IO/Ports/SerialPort.OSX.cs

```diff
@@ -0,0 +1,36 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.ComponentModel;
+using System.Collections.Generic;
+using System.IO;
+
+namespace System.IO.Ports
+{
+    public partial class SerialPort : Component
+    {
+        public static string[] GetPortNames()
+        {
+            List<string> ports = new List<string>();
+
+            foreach (string name in Directory.GetFiles("/dev", "tty.*"))
+            {
+                // GetFiles can return unexpected results because of 8.3 matching.
+                if (name.StartsWith("/dev/tty."))
```

Comment:
I don't think you need this extra check here, nor the one in the other loop. If you don't need them you also don't need the ports list you can just create the array and put the entries.

Reply:
I think I do. Without that check I was getting "/dev/tty" and that is not what I want.
Per the comment, it seems to be expected per MsDos compatibility ;(